1-ECommerce/src/App.tsx

```diff
@@ -1,14 +1,35 @@
 import "./index.css";
 
+import type { Dispatch, SetStateAction } from "react";
+import { createContext } from "react";
+import { useSearchParams } from "react-router-dom";
+
 import { NavBar } from "./components/NavBar";
 import { ProductsContainer } from "./components/ProductsContainer";
 
+interface SearchParamContext {
+  queryParams: URLSearchParams;
+  setQueryParams: Dispatch<SetStateAction<URLSearchParams>>;
+}
+
+export const QueryParamContext = createContext({} as SearchParamContext);
+
 export default function App() {
+  const [queryParams, setQueryParams] = useSearchParams();
```

Comment:
I don't think there is a need for this Context, doesn't `useSearchParams` already have the same values everywhere you use it anyways?

What I mean is that if you just do this in any other component, you would get the same `queryParams` and `setQueryParams`:

```
const [queryParams, setQueryParams] = useSearchParams();
```

Reply:
That's a good observation. I'm not sure if removing the context and simply using `useSearchParams` in the components would register changes that were generated from other components. I'll try it and see.